docs/source/configuration/slots.md

```diff
@@ -0,0 +1,379 @@
+---
+myst:
+  html_meta:
+    "description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:title": "Slots"
+    "keywords": "Volto, Plone, frontend, React, configuration, slots, viewlets"
+---
+
+# Slots
+
+Slots provide a way for Volto add-ons to insert their own components at predefined locations in the rendered page.
+
+```{note}
+This concept is inspired by the Plone Classic UI {doc}`plone:classic-ui/viewlets`.
+```
+
+
+## Anatomy
+
+Slots have a name, and they contain a list of slot components.
+
+Volto renders slots using the `SlotRenderer` component.
+You can add slot insertion points in your code, as shown in the following example.
+
+```ts
+<SlotRenderer name="toolbar" content={content} />
+```
+
+Slot components consist of a data structure with key/value pairs, where the keys are the parent's slot name, the slot component's name, the component to render in the slot, and optional predicates.
+They are registered in the {ref}`configuration registry using a specific API for slots <configuration-registry-for-slot-components>`.
+
+The renderer of a slot component is controlled by the presence or absence of a list of conditions called {term}`predicates`.
+
+You can register multiple slot components with the same name under the same slot, as long as they have different predicates or components.
+
+To illustrate how slots are structured and work, let's register a slot component, where the component is `@sneridagh what goes here?`, and the predicate matches a route that begins with `/de/about`.
+
+```ts
+config.registerSlotComponent({
+  slot: 'toolbar',
+  name: 'save',
+  component: '@sneridagh what goes here?',
+  predicates: [RouteCondition('/de/about')],
```

Comment:
@stevepiercy The example of changing the save button in the toolbar for different predicates seems a bit contrived. How about adding a different header to the aboveContent slot? So the slot would be `aboveContent`, name would be `header`, component would be a hypothetical React component AboutHeader, and the predicates would remain the same.

Reply:
Agree. I had the idea of an alert that you would post sitewide on a university website for a campus shooter, wildfire, or ice storm, but that was depressingly too realistic and would not require predicates. Your example could apply for departments or other organizations in a university. I'll run with that.

Reply:
Done in [40c1dfd](https://github.com/plone/volto/pull/5775/commits/40c1dfd3125bb2ee1594bffcef8560948bbc4130)